src/AspNetCoreModuleV2/AspNetCore/Inc/applicationinfo.h

```diff
@@ -10,7 +10,7 @@ typedef
 HRESULT
 (WINAPI * PFN_ASPNETCORE_CREATE_APPLICATION)(
     _In_  IHttpServer        *pServer,
-    _In_  ASPNETCORE_CONFIG  *pConfig,
+    _In_  IHttpContext       *pHttpContext,
```

Comment:
Doesn't seem right, isn't `IHttpContext` per request? Why would app need it in addition to `IHttpServer`?

Reply:
you need IHttpServer to be able to read configuration. You also need IIS application info which is available at httpcontext. I agree it looks wired to pass httpcontext directly. We should only pass necessary info

Reply:
I'll need to modify my websocket PR to set whether websockets are enabled later on as I use the HttpContext for it. Otherwise, I can change the IHttpContext to IHttpApplication.